### DISCUSSION TITLE
Sensor board for compass face

### DISCUSSION THREAD

**Discussion by kyleboyle** [joeycastillo/Sensor-Watch#143] [Q&A]
Hi there, I was wondering what sensor board I'd need to enable a compass face?

**Comment by joeycastillo** [comment]
I’ve never done a compass before, but a magnetometer would be the kind of sensor you’d need! I would imagine the [LIS2MDL](https://www.digikey.com/en/products/detail/stmicroelectronics/LIS2MDLTR/7348328) as being a good choice, but before you commit to a sensor board with it, I’d recommend getting one of [these breakouts](https://www.adafruit.com/product/4488) and something like a QT Py to play with it and see if you can get the data you want out of it. If you find you can find magnetic north with it, the next step would be to put it on a sensor board!

**Reply by kyleboyle** [reply]
Cool thanks for the pointers. I've not done this type of development before and I'm a long time removed from writing embedded c code for tv set top boxes.

**Comment by kyleboyle** [comment]
@joeycastillo Is the power consumption for something like the [LSM303AH](https://www.st.com/en/mems-and-sensors/lsm303ah.html#documentation) too high... it lists current consumption in power-down at 2.5μA while the lis2dw is at 50nA. 
The LSM303 will give built in pedometer, tap and double tap gestures... as well as being able to use the accel and magnetometer for tilt corrected compass headings.

**Reply by joeycastillo** [reply]
2.5µA is workable, but alas there's another line in that data sheet that makes it unlikely to work: the LSM303AH is 1.8V power and logic. Since our coin cell is 3V, you'd need to add both a 1.8V regulator and at least two signals worth of level shifting to get it to work, which I think might be troublesome to fit on that tiny sensor board. 

I hit a similar wall with the [ICM-20948](https://invensense.tdk.com/wp-content/uploads/2016/06/DS-000189-ICM-20948-v1.3.pdf); it seems like 1.8V is a trend with some of these newfangled sensors, and it's going to be a challenge to use any of them in Sensor Watch.

**Reply by kyleboyle** [reply]
How about the [lsm303agr](https://www.st.com/en/mems-and-sensors/lsm303agr.html#documentation) it looks like it satisfies the 3v requirement. It lists current consumption in power-down at 2μA. I got one of the adafruit breakout boards for it and got a tilt compensated compass heading running in python on a raspberrypi. The only functional difference I can see is that it doesn't have the built in pedometer / step counter compared to AH.

**Reply by joeycastillo** [reply]
That may be the winner! And the current consumption looks good. Just for a general sense of power budget, wake mode consumes ~6µA and sleep mode ~4µA. This would bump it up to 8µA and 6µA respectively, and in my experience if you can keep average power consumption below 10µA you're likely to hang on to that year-long battery life. 

Only other note I'd say is I'd recommend putting the accelerometer interrupts on pins A2 and A4, as these are the pins that can wake the watch up from its low power modes. This could be a good way to explore things like wake on wrist raise, or entering a deeper sleep mode when the watch is sitting still for a long time.

**Reply by Jonas-fr** [reply]
> The LSM303 will give built in pedometer, tap and double tap gestures...

That reminds me of an old Nokia phone of mine which had  a nifty functionality I never found on another phone. When sleeping (screen off), a double tap on the phone would display briefly the time on the screen. I wonder whether such a functionality could be added to the main face (or at least the regular clock face). It would allow the screen to be off in deep sleep mode which in turns would improve the battery life considerably (giving we'd use a sensor which doesn't consume much).

**Comment by siggifromgermany** [comment]
Hi, I just ordered the Sensor-Watch a few days ago and have zero experiences with it so far (it´s not here yet). But today I was in the forest by bike and thought about how nice it would be, if there was a possibility to use the Watch also as a compass.
So I would be pleased to hear what the current status with this project is?

Greetings from Germany

**Reply by kyleboyle** [reply]
I agree it would be nice. My discussion here was just curiosity. I'm not familiar with creating hardware and don't know how to create a sensor board.

**Comment by fschuele** [comment]
Hi,
I had a similer idea as @siggifromgermany and was wondering if an HMC5883L would do the job? No need for accelerometer or other fancy stuff.

From the datasheet:
- SPI/I2C
- 1.71 - 3.6 V
- 100 uA (Measurement mode)
- 2 uA (Idle mode)
- 3.0x3.0x0.9mm

Measurement mode could be switched on only when a button is/was pressed or something like that?

I really wonder if the following could be a major issue to such a project (from the datasheet "_LAYOUT CONSIDERATIONS_" of the HMC):

_Besides keeping all components that may contain ferrous materials (nickel, etc.) away from the sensor on both sides of
the PCB it is also recommended that there is no conducting copper under/near the sensor in any of the PCB layers._

Do you guys think it would be possible to place a sensor board containing the HMC away from ferrous material of the watch? Does
the watch even contain ferrous material?
